_includes/includes/com.keyman.help.editpage.inc.php

```diff
@@ -1,69 +1,59 @@
 <?php
   namespace com\keyman\help\editpage;
 
+  use \Keyman\Site\Common\KeymanVersion;
+
+  // TODO: it may be better to have a file in the root of each folder which
+  //       determines the root path for editing and any parameters relating
+  //       to it.
+
   function getPageUrlForEditLink($self, $id, $file) {
     $base = "https://github.com/keymanapp/help.keyman.com/edit/master";
     $app =  "https://github.com/keymanapp/keyman/edit/master";
 
     $map = [
-      'android' => 'android/docs/help/',
-      'iphone-and-ipad' => 'ios/docs/help/',
-      'linux' => 'linux/docs/help/',
-      'mac' => 'mac/docs/help/',
-      'windows' => 'windows/docs/help/'
+      'products/android' => 'android/docs/help/',
+      'products/iphone-and-ipad' => 'ios/docs/help/',
+      'products/linux' => 'linux/docs/help/',
+      'products/mac' => 'mac/docs/help/',
+      'products/windows' => 'windows/docs/help/',
+      'developer' => 'developer/docs/help/',
+      'developer/engine/android' => 'android/docs/engine/',
+      'developer/engine/iphone-and-ipad' => 'ios/docs/engine/',
+      'developer/engine/web' => 'web/docs/engine/',
+      'developer/engine/windows' => 'windows/docs/engine/',
     ];
-    $maps = join('|', array_keys($map));
+    $maps = preg_replace('/\//', '\\/', join('|', array_keys($map)));
 
     if($self == '/knowledge-base/index.php' && !empty($id)) {
       // Knowledge Base Article
       require_once('includes/com.keyman.help.kb.inc.php');
       return $base."/knowledge-base/".\com\keyman\help\kb\filename_from_id($id);
     }
-    else if(preg_match('/^products\/('.$maps.')\/(\d+\.\d+)\/(.*)\.md$/', $file, $matches) && version_compare($matches[2], '14.0', '>=')) {
-      // Product help, version 14.0 and later
+    else if(preg_match('/^('.$maps.')\/(\d+\.\d+)\/(.*)\.md$/', $file, $matches) && version_compare($matches[2], KeymanVersion::stable_version, '>=')) {
+      // Product, Developer, Engine help, version 14.0 and later. Note that this
+      // is not a perfect match and in some cases will have missing links
+      // because we are not attempting to check for latest version of help
       return "$app/{$map[$matches[1]]}{$matches[3]}.md";
     }
+    else if(preg_match('/^(products\/|developer\/engine\/.+|(developer\/(\d+\.\d+)))/', $file, $matches)) {
```

Comment:
Can't we use the same pattern as above?

```suggestion
    else if(preg_match'/^('.$maps.')\/(\d+\.\d+)\/(.*)\.md$/', $file, $matches)) {
```

Reply:
This is a broader match -- also captures index pages, etc. Will add a comment.